index.js

```diff
@@ -49,51 +47,28 @@ if (!config) {
   throw new Error('you did\'t define any krabby reports. great job.');
 }
 
-tests = config.tests.map(function(test) {
-  var testName;
-  var config = {};
+var instantiatePlugins = function(plugins, path, rootMethod) {
+  return plugins.map(function(plugin) {
+    var pluginConfig = _.isString(plugin) ? { name: plugin } : plugin;
 
-  if (typeof test === 'string') {
-    testName = test;
-  } else {
-    testName = test.name;
-
-    config = test;
-  }
-
-  var Test = require(Path.join(__dirname, 'lib/tests', testName));
-
-  test = new Test(config);
-
-  return test.test.bind(test);
-});
-
-reports = config.reports.map(function(report) {
-  var reportName;
-  var config = {};
-
-  if (typeof report === 'string') {
-    reportName = report;
-  } else {
-    reportName = report.name;
-    config = report;
-  }
-
-  report = require(Path.join(__dirname, 'lib/reports', reportName));
-  report = report(config);
-
-  return report;
-});
+    var Plugin = require(Path.join(__dirname, path, pluginConfig.name));
+    var pluginInstance = new Plugin(pluginConfig);
 
+    return pluginInstance[rootMethod].bind(pluginInstance);
```

Comment:
We could make this even more simple by figuring out what a good method name should be for both tests AND reports